The art of drunken captaining

Code-Cygnet opened the pull request:
Drinking on the job is good.

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This PR is basically an attempt at adding a special drink for captains called "Commander and Chief".


## Why It's Good For The Game

Some departments have a special drink that improves the performance of those within them. Life on the station could get rough for a Captain, who oversees the entire station. When times get hard enough that he/she/they might need a drink, they could order a Commander and Chief from the bar before rolling up her/his/their sleeves to put in some work. 

## Changelog
:cl:
add: Added new things - Mind trait, alcohol reagent, chemical reagent, drink sprite and recipe.
imageadd: added commander_and_chief sprite to drinks.dmi
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Code-Cygnet committed:
Adds new drink to drink recipes

It's a drink for Captains

Code-Cygnet committed:
Uploads new drinks.dmi 

Adds Commander and Chif

Code-Cygnet committed:
Update traits.dm

Adds new mob trait at line 121

Code-Cygnet committed:
Update captain.dm

Adds mind trait at line 24

Code-Cygnet committed:
Update alcohol_reagents.dm

Adds commander and chief, starting at line 2125 and ending at 2146.. hopefully.

Code-Cygnet committed:
Update captain.dm

Attempted error fix.

Code-Cygnet committed:
Merge branch 'master' into The-Art-of-Drunken-Captaining

Code-Cygnet committed:
Update alcohol_reagents.dm

Edited a typo at line 2144.

Code-Cygnet committed:
Update captain.dm

Sishen1542 commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2122,6 +2122,29 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	glass_desc = "The gems of this majestic chalice represent the departments and their Heads."
+	
+/datum/reagent/consumable/ethanol/commander_and_chief/on_mob_life(mob/living/carbon/M)
+	if(M.mind && HAS_TRAIT(M.mind, TRAIT_CAPTAIN_METABOLISM)))
```
```suggestion
	if(M.mind && HAS_TRAIT(M.mind, TRAIT_CAPTAIN_METABOLISM))
```

Should fix your compile issues.

Code-Cygnet committed:
Update code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

Error spotted at 2138 and fixed by Sishen, aka Papa Pax!

Co-Authored-By: Sishen1542 <sishen1542@gmail.com>

Code-Cygnet committed:
Update drinks_recipes.dm

Fixed error at line 803, allies_cocktail to alliescocktail. Am bone-head :(.

Sishen1542 commented on code/modules/food_and_drinks/recipes/drinks_recipes.dm:
```diff
@@ -794,4 +794,11 @@
+	name = "Commander and Chief" 
+	id = "commander_and_chief" 
+	results = list("commander_and_chief" = 50) 
+	required_reagents = list("allies_cocktail" = 50, "champagne" = 20, "doctors_delight" = 10, "quintiple_sec" = 10, "screwdriver" = 10)
```
```suggestion
	required_reagents = list("alliescocktail" = 50, "champagne" = 20, "doctorsdelight" = 10, "quintuple_sec" = 10, "screwdrivercocktail" = 10)
more fixies

Code-Cygnet replied:
Heck the only one I had right was "champagne" LOL. Thanks for the help :D.

Sishen1542 replied:
no problemo, gotta start somewhere :heart:

Code-Cygnet committed:
Update drinks_recipes.dm

Fixed more errors, spelling and id errors to be exact with the help of Papa Pax :D.
